scripts/tools/SetStateStorageForTools.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoQtiTest\scripts\tools;
+
+use common_report_Report as Report;
+use oat\oatbox\extension\AbstractAction;
+use oat\taoQtiTest\models\runner\toolsStates\KvToolsStateStorage;
+use oat\taoQtiTest\models\runner\toolsStates\NoStorage;
+use oat\taoQtiTest\models\runner\toolsStates\RdsToolsStateStorage;
+use oat\taoQtiTest\models\runner\toolsStates\ToolsStateStorage;
+
+/**
+ * Register given implementation of 'ToolsStateStorage' and prepares the storage, if needed
+ *
+ * Class InstallRdsToolsStateStorage
+ * @package oat\taoQtiTest\scripts\install
+ */
+class SetStateStorageForTools extends AbstractAction
+{
+    /**
+     * @param array $params
+     * @return Report
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     */
+    public function __invoke($params)
+    {
+        if (count($params) === 0) {
+            return new Report(Report::TYPE_ERROR, 'Storage type is not provided');
+        }
+
+        $storageType = $params[0];
+
+        switch ($storageType) {
```

Comment:
Maybe we can clean this up by dividing the switch cases into seperate methods, eg:

`$this->registerRdsStorage()`
`$this->registerKeyValueStorage()`
`$this->registerNoStorage()`